src/clp_ffi_js/ir/StructuredIrUnitHandler.cpp

```diff
@@ -274,4 +291,44 @@ auto StructuredIrUnitHandler::get_timestamp(StructuredLogEvent const& log_event)
             timestamp.get_immutable_view<clp::ffi::value_int_t>()
     );
 }
+
+auto StructuredIrUnitHandler::get_utc_offset(StructuredLogEvent const& log_event) const
+        -> UtcOffset {
+    constexpr std::chrono::minutes cDefaultUtcOffset{0};
+
+    if (false == m_optional_utc_offset_full_branch.has_value()) {
+        return cDefaultUtcOffset;
+    }
+
+    if (false == m_optional_utc_offset_node_id.has_value()) {
+        return cDefaultUtcOffset;
+    }
+
+    auto const utc_offset_node_id = m_optional_utc_offset_node_id.value();
+    auto const& node_id_value_pairs = m_optional_utc_offset_full_branch->is_auto_generated()
+                                              ? log_event.get_auto_gen_node_id_value_pairs()
+                                              : log_event.get_user_gen_node_id_value_pairs();
+    if (false == node_id_value_pairs.contains(utc_offset_node_id)) {
+        return cDefaultUtcOffset;
+    }
+
+    auto const& optional_utc_offset_value = node_id_value_pairs.at(utc_offset_node_id);
+    if (false == optional_utc_offset_value.has_value()) {
+        SPDLOG_ERROR(
+                "Protocol error: The utc offset cannot be an empty value. Log event index: {}",
+                m_deserialized_log_events->size()
+        );
+        return cDefaultUtcOffset;
+    }
+
+    auto const& utc_offset_value{optional_utc_offset_value.value()};
+    if (false == utc_offset_value.is<clp::ffi::value_int_t>()) {
+        SPDLOG_ERROR(
+                "Protocol error: The utc offset must be a valid integer. Log event index: {}",
+                m_deserialized_log_events->size()
+        );
+        return cDefaultUtcOffset;
+    }
+    return UtcOffset{utc_offset_value.get_immutable_view<clp::ffi::value_int_t>()};
```

Comment:
```suggestion
    // TODO: Support parsing of other time units, e.g., minutes, hours, etc..
    auto const utc_offset_seconds
            = std::chrono::seconds{utc_offset_value.get_immutable_view<clp::ffi::value_int_t>()};
    return std::chrono::duration_cast<UtcOffset>(utc_offset_seconds);
```

need to tell chrono that the input unit is seconds